app/src/editor/canvas/components/Resizable/index.jsx

```diff
@@ -121,6 +126,36 @@ const Resizable = ({
         })
     }, [width, height])
 
+    const updatePreviousSize = useCallback(() => {
+        previousSize.current = {
+            height,
+            width,
+        }
+    }, [width, height])
+
+    const updatePreviousSizeRef: Ref<Function> = useRef(updatePreviousSize)
+
+    useEffect(() => {
+        updatePreviousSizeRef.current = updatePreviousSize
+    }, [updatePreviousSize])
+
+    const commitRef: Ref<Function> = useRef(commit)
+
+    useEffect(() => {
+        commitRef.current = commit
+    }, [commit])
```

Comment:
I think you could/should just set this in the render, rather than inside an effect?

i.e.

```js
const commitRef: Ref<Function> = useRef()
commitRef.current = commit
```

Same for `updatePreviousSize`

Reply:
Also means if these effects happen to get re-ordered to come after the updatePreviousSize/commit effect below, then it'll be quietly calling a stale callback which seems bad. If you set the ref in the render body then there's no way the effect can possibly run before the ref is updated.